doc/source/dtypes.rst

```diff
@@ -0,0 +1,121 @@
+.. currentmodule:: geopandas
```

Comment:
I wouldn't call this `dtypes.rst`, as 'dtype' has already a distinct meaning in the numpy/pandas world, and which is something else as discussed in this file